noxfile.py

```diff
@@ -819,28 +830,87 @@ def notebook(session: nox.Session):
+
+
+def _process_benchmark_recursively(
+    session: nox.Session,
+    current_path: Path,
+    benchmark_configs: List[Tuple[Optional[str], List[str]]] = [(None, [])],
+):
+
+    config_path = current_path / "config.jsonl"
+
+    if config_path.exists():
+        benchmark_configs = []
+        with open(config_path, "r") as f:
+            for line in f:
+                config = json.loads(line)
+                python_args = [
+                    f"--{key}={value}"
+                    for key, value in config.items()
+                    if key != "benchmark_suffix"
+                ]
+                suffix = (
+                    config["benchmark_suffix"]
+                    if "benchmark_suffix" in config
+                    else "_".join(f"{key}_{value}" for key, value in config.items())
+                )
+                benchmark_configs.append((suffix, python_args))
+
+    benchmark_script_list = list(current_path.glob("*.py"))
     processes = []
     for benchmark in benchmark_script_list:
-        process = Process(
-            target=session.run,
-            args=("python", benchmark),
-            kwargs={"env": {LOGGING_NAME_ENV_VAR: benchmark.as_posix()}},
+        if benchmark.name == "utils.py":
+            continue
+
+        for benchmark_config in benchmark_configs:
+            args = ["python", benchmark]
+            args.extend(benchmark_config[1])
+
+            log_env_name_var = benchmark.as_posix()
+            if benchmark_config[0] is not None:
+                log_env_name_var += f"_{benchmark_config[0]}"
+            kwargs = {"env": {LOGGING_NAME_ENV_VAR: log_env_name_var}}
+
+            process = Process(
+                target=_benchmark_timing_process,
+                args=(session, args, kwargs),
+            )
+            process.start()
+            processes.append(process)
+
+    for sub_dir in [d for d in current_path.iterdir() if d.is_dir()]:
+        processes.extend(
+            _process_benchmark_recursively(session, sub_dir, benchmark_configs)
         )
-        process.start()
-        processes.append(process)
 
-    for process in processes:
-        process.join()
+    return processes
 
-    # when the environment variable is set as it is above,
-    # notebooks output a .bytesprocessed and .slotmillis report
-    # collect those reports and print a summary
-    _print_performance_report(base_path)
+
+def _benchmark_timing_process(session: nox.Session, args, kwargs, location=None):
+    clock_time_file_path = (
+        f"{location or kwargs['env'][LOGGING_NAME_ENV_VAR]}.clockseconds"
+    )
+    start_time = time.perf_counter()
+    benchmark_proc = Process(
+        target=session.run,
+        args=args,
+        kwargs=kwargs,
+    )
+    benchmark_proc.start()
+    benchmark_proc.join()
+    end_time = time.perf_counter()
+    runtime = end_time - start_time
+    with open(clock_time_file_path, "w") as log_file:
+        log_file.write(f"{runtime}\n")
```

Comment:
There's a lot of overhead in starting up a Python process and authenticating. I don't think this metric would be particularly useful for what we need it to be. Can we do this wall clock timing in the benchmark script itself?

Reply:
Yes, it is possible, but in that case the metric won't work for notebooks. unless we add timing to notebooks too. Remove this for now.

Reply:
Updated to get time from benchmark script,  and for notebook there is one less metric.